test/storage/tile_compression_test.cpp

```diff
@@ -0,0 +1,181 @@
+  catalog::Column column1(type::TypeId::INTEGER, type::Type::GetTypeSize(type::TypeId::INTEGER),
+                          "A", true);
+  catalog::Column column2(type::TypeId::VARCHAR, 25, "B", false);
+  catalog::Column column3(type::TypeId::INTEGER, type::Type::GetTypeSize(type::TypeId::INTEGER),
+                          "C", true);
+  catalog::Column column4(type::TypeId::TINYINT, type::Type::GetTypeSize(type::TypeId::TINYINT),
+                          "D", true);
+  catalog::Column column5(type::TypeId::VARCHAR, 25, "E", false);
+
+  columns.push_back(column1);
+  columns.push_back(column2);
+  columns.push_back(column3);
+  columns.push_back(column4);
+  columns.push_back(column5);
+
+  // Schema
+  std::unique_ptr<catalog::Schema> schema(new catalog::Schema(columns));
+
+  // Allocated Tuple Count
+  const int tuple_count = 6;
+
+  std::unique_ptr<storage::TileGroupHeader> header(
+      new storage::TileGroupHeader(BackendType::MM, tuple_count));
+
+  std::unique_ptr<storage::Tile> tile(storage::TileFactory::GetTile(
+      BackendType::MM, INVALID_OID, INVALID_OID, INVALID_OID, INVALID_OID,
+      header.get(), *schema, nullptr, tuple_count));
+
+  std::unique_ptr<storage::Tuple> tuple1(new storage::Tuple(schema.get(),
+                                                            true));
+  std::unique_ptr<storage::Tuple> tuple2(new storage::Tuple(schema.get(),
+                                                            true));
+  std::unique_ptr<storage::Tuple> tuple3(new storage::Tuple(schema.get(),
+                                                            true));
+  auto pool = tile->GetPool();
+
+  tuple1->SetValue(0, type::ValueFactory::GetIntegerValue(11), pool);
+  tuple1->SetValue(1, type::ValueFactory::GetVarcharValue("bohan"), pool);
+  tuple1->SetValue(2, type::ValueFactory::GetIntegerValue(12), pool);
+  tuple1->SetValue(3, type::ValueFactory::GetTinyIntValue(13), pool);
+  tuple1->SetValue(4, type::ValueFactory::GetVarcharValue("bohan"),
+                   pool);
+
+  tuple2->SetValue(0, type::ValueFactory::GetIntegerValue(21), pool);
+  tuple2->SetValue(1, type::ValueFactory::GetVarcharValue("ssy"), pool);
+  tuple2->SetValue(2, type::ValueFactory::GetIntegerValue(22), pool);
+  tuple2->SetValue(3, type::ValueFactory::GetTinyIntValue(23), pool);
+  tuple2->SetValue(4, type::ValueFactory::GetVarcharValue("ssy again"), pool);
+
+  tuple3->SetValue(0, type::ValueFactory::GetIntegerValue(31), pool);
+  tuple3->SetValue(1, type::ValueFactory::GetVarcharValue("tao dai"), pool);
+  tuple3->SetValue(2, type::ValueFactory::GetIntegerValue(32), pool);
+  tuple3->SetValue(3, type::ValueFactory::GetTinyIntValue(33), pool);
+  tuple3->SetValue(4, type::ValueFactory::GetVarcharValue("tao dai again"), pool);
+
+  tile->InsertTuple(0, tuple1.get());
+  tile->InsertTuple(1, tuple2.get());
+  tile->InsertTuple(2, tuple3.get());
+  
+  // encode the tile
+	std::unique_ptr<storage::DictEncodedTile> encodedTile(
```

Comment:
std::make_unique instead of new -- this applies to pretty much all uses of new in this test.